File: src/main/java/br/com/matheuspadilha/tasks/model/Address.java
package br.com.matheuspadilha.tasks.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serial;
import java.io.Serializable;

@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Address implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    @JsonProperty("cep")
    private String zipCode;

    @JsonProperty("logradouro")
    private String street;

    @JsonProperty("complemento")
    private String complement;

    @JsonProperty("bairro")
    private String neighborhood;

    @JsonProperty("localidade")
    private String city;

    @JsonProperty("uf")
    private String state;

}

File: src/main/kotlin/br/com/matheuspadilha/tasks/model/Address.kt
package br.com.matheuspadilha.tasks.model

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonInclude.Include
import com.fasterxml.jackson.annotation.JsonProperty
import java.io.Serializable

@JsonInclude(Include.NON_NULL)
data class Address(
    @JsonProperty("cep")
    val zipCode: String? = null,

    @JsonProperty("logradouro")
    val street: String? = null,

    @JsonProperty("complemento")
    val complement: String? = null,

    @JsonProperty("bairro")
    val neighborhood: String? = null,

    @JsonProperty("localidade")
    val city: String? = null,

    @JsonProperty("uf")
    val state: String? = null,
) : Serializable {}
